### DISCUSSION TITLE
MiniPhonie / TravelPhonie

### DISCUSSION THREAD

**Discussion by apkuki** [MiczFlor/RPi-Jukebox-RFID#1860] [Show and tell]
Hi together
As my son (turning 3) really loves his Phoniebox and wants to take it with him to travel I thought I'll give it a go and try to build a "minified version". As I know many of you opt for a "smaller" Version of your Box. I'll share my journey so far.

### The Parts
Stumbling upon [PiSugar   ](https://www.pisugar.com/) I started the project with the following parts list

- Raspberry Pi Zero 2 W
- Pisugar3
- Pirate Audio Headphone Amp
- [Seeed RFID Reader](https://www.berrybase.ch/seeed-grove-125khz-rfid-leser)

I soon run into troubles because the Seed RFID Reader even though it is super small puts out 5V (Arduino) and not 3V for the GPIO of a Raspberry. I thought about getting a Logic Level Converter - but that would compromise with my minified approach.

As I already worked with the Neuftech/Yarontech USB Reader in my first Box I knew that the PCB is quite small. I therefore ordered another Neuftech/Yarongtech USB eager to modify it to work with my RaspberryPi Zero 2

![Reader Side View](https://user-images.githubusercontent.com/98207992/180413732-f4122491-e01f-4301-85c8-9a12ed59e2e1.jpg)
![Reader Up View](https://user-images.githubusercontent.com/98207992/180413741-d3a9c638-8259-49db-a078-1221b2984aeb.jpg)

As you see in the Up View the PCB of the Neuftech Reader on the right is larger and also the Antenna is way bigger than the one from the Seeed on the left.

Compared from the side though you see that on the Neuftech on the right only the "Buzzer" and the MiniUSB Stick out and that the Seed on the left has more components sticking up.

### The Build
I therefore combined the positive things of both readers and altered the PCB of the Neuftech Reader as folows:
- getting rid of the buzzer (no problem as its not a vital component)
- getting rid of the MiniUSB Port
- soldering USB 5V, Ground, D+ and D- directly to the PCB on the given solder points
- attaching the smaller Seeed Antenna to the Neuftech PCB
 
![New Reader Side](https://user-images.githubusercontent.com/98207992/180415125-ed5fad88-1d2f-432f-b55f-45652d0c3848.jpg)
![New Reader Up](https://user-images.githubusercontent.com/98207992/180415137-ae3957af-9e63-422c-8827-46455cf9732a.jpg)
 
So overall you win quite a lot of space. You loose the buzzer beeping but as it will be used with headphones it's not really relevant and you still got the LED of the reader lighting up green when reading a card.

For testing purposes I attached a MicroUSB Plug to my cables and plugged it into the PiZero -> It works perfect!
The range of the smaller antenna is not as wide as the neuftech (maybe around 3cm) for that minified version it's perfectly fine though. 

The Phoniebox Software and the Pirate Audio behaved a little unwell on the Raspberry Pi Zero 2. So I actually had to install the Pirate Audio first with all its Mopidy components and software via the Pimoroni Install: https://learn.pimoroni.com/article/getting-started-with-pirate-audio and then I installed the RPI-Jukebox Solution in Developer Version: https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/INSTALL-stretch keep in mind that you need to install the Spotify Version without MPD. There may be other working combinations though

### Prototype and Size
![Components together](https://user-images.githubusercontent.com/98207992/180416519-1af33e00-b3fc-4d60-9f23-a8dc564da0e1.jpg)
![Stacked Side](https://user-images.githubusercontent.com/98207992/180416509-f12fb628-a597-4148-9c86-465cbca6ada1.jpg)
![Stacked Up](https://user-images.githubusercontent.com/98207992/180416513-8cfed9c4-fe69-4058-9e42-1ffc6b90c7e3.jpg)

Overall compared to a swiss army knife without the reader:
![Size Comparison UP](https://user-images.githubusercontent.com/98207992/180417217-d9af92f5-9bbd-4a58-8717-2aaaa2494b0b.jpg)
![Size Comparison Side](https://user-images.githubusercontent.com/98207992/180417223-0202ceeb-268a-4b05-9af1-18fe61b0440c.jpg)

### What's still to do
So next step would be to "finish" the whole package. I will therefore solder the USB cables directly to the RaspberryPi Zero 2 using the ["TestPads" underneath](https://datasheets.raspberrypi.com/rpizero2/raspberry-pi-zero-2-w-test-pads.pdf) 
I know they haven't been designed for that - but as many others have actually used the pads for attaching USB Devices to the Pi Zero 2 via the pads I'll give it a go to keep it as minified as possible.

**3d Printing**
For me it will be a first to 3D Print an enclosure. I already started an approach with a combination of Templates from the PiSugar [Github](https://github.com/PiSugar/PiSugar/tree/master/model3) and some PirateAudio STL's I found online.
Still need to figure out something for the Reader PCB which will be on the side probably. As this PCB has 4 hole I'll try print something to stick out so that it will hold the reader and antenna in place.
First STL File: [Miniphonie.zip](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/9166899/Miniphonie.zip)

If somebody has more experience in 3d printing and measure to print... let my know I'm really open for suggestions from more experienced users!

Let me know in the comments if you know any other hardware or approaches when it would come to minify - especially the readers and antennas 👍

**Comment by apkuki** [comment]
A first little update:

The "Miniphonie" works perfectly well - i used Splitt's Synchronizing Script https://splittscheid.de/phoniebox-synchronisieren/ (German) to synchronize the main Phoniebox and the Miniphonie. After that all cards could be played easily on the Miniphonie.
The startuptime is 1min 24s from the push of the button to the startup sound (Mopidy.Server: Connected). Surely something I have to look into it in the future (again checking Splitti's optimizing Blog https://splittscheid.de/phoniebox-schneller-booten-fehler-beheben/ , but for now it is OK - and I'm happy everything worked out as planned.


**RFID Reader:**
The Reader is now soldered directly to the RPi Zero 2 - it works perfectly fine.
![PXL_20220810_094200578](https://user-images.githubusercontent.com/98207992/183928969-8b14019c-f42f-4e9b-9329-967e74fc14bb.jpg)
as its is added on the side i taped the soldering points with pcb tape just to be shure to not have any shorts
![PXL_20220810_094434798](https://user-images.githubusercontent.com/98207992/183929281-76e55e43-e3d7-4239-986f-1c591de8fea7.jpg)


**The Miniphonie**
After the Reader has been soldered to the Pi 2 Zero I screwed the whole package together. It fits nicely toghether and the reader is almost the same size as the whole package.
The length is 68.5mm including the SD Card
The width is 35.6mm including the reader and antenna
The height is 32.9mm including the antenna

![PXL_20220810_135140560](https://user-images.githubusercontent.com/98207992/183928037-8caed53b-c896-4ce7-b867-9c32a1729c7a.jpg)
![PXL_20220810_135207511](https://user-images.githubusercontent.com/98207992/183928083-8a9dc27b-4c64-4429-aad3-a1180faf2984.jpg)
![PXL_20220810_135235065](https://user-images.githubusercontent.com/98207992/183928119-52749b6f-3932-4b9f-a5cf-7501953c05d4.jpg)


What's still to to:
Design and print a 3d enclosure --> if anybody is skilled in 3D Modelling and is eager to help - let me know.

**Comment by lh84** [comment]
Hi, can you say something how long the battery of the pisugar lasts? Thanks in advance

**Comment by d4rkd3v1l** [comment]
We already built two phonieboxes for our kids. However they are not quite portable and so we thought about building something smaller for travel etc. 
Doing some research led us to this discussion. This is already like 3,5 years old, but I wonder what became of this project. So @apkuki, if you read this, maybe you'd like to share some more information about how this turned out in the end :-) Thanks! :-)
